### PR TITLE
[Statsig] Add Datadog-Statsig Integration

### PR DESCRIPTION
**Pull request by marcos-statsig**
### What does this PR do?
Adds the initial Statsig integration listing

### Motivation
Adds integration listing for Statsig to enable Statsig users to be able to monitor their feature gate changes and rollouts in Datadog.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes
N/A

### CODE REVIEW

**Review by jtappa — APPROVED** [review]
👍🏻 From docs

**Review by FlorianVeaux — CHANGES_REQUESTED** [review]
Looks good, thanks for providing some logos but you can remove them from the PR. It doesn't need to be included in this repository.
Besides that and my comment below, this is good to go.

File: statsig/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+4. Click on the Datadog card.
+5. Paste in your API key in the top field and click Confirm.
+
+## Data Collected
+
+The Statsig integration does not collect any data from Datadog.
+
+### Metrics
+
+See [metadata.csv][] for a list of metrics provided by this integration and the description of each.
```
[the remark this answers is not in the document]
Comment (marcos-statsig) [reply statsig/README.md]: Great, thank you for the review!

**Review by FlorianVeaux — APPROVED** [review]
Looks great, going to merge that later today 👍

File: statsig/manifest.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  "metric_to_check": "statsig.log_event.count",
+  "creates_events": true,
+  "short_description": "Monitor Statsig changes in Datadog",
+  "guid": "f22e08ca-b524-45ad-815c-9ad654015158",
+  "support": "contrib",
+  "supported_os": ["linux", "mac_os", "windows"],
+  "public_title": "Statsig",
+  "categories": [""],
+  "type": "check",
+  "is_public": false,
```
Comment (FlorianVeaux) [inline statsig/manifest.json:16]:
```suggestion
  "is_public": true,
```

**Review by FlorianVeaux — CHANGES_REQUESTED** [review]
One more thing, set `is_public` to true to make your integration discoverable

**Review by FlorianVeaux — APPROVED** [review]

**Pull request merged**